### DISCUSSION TITLE
Buttons in labels

### DISCUSSION THREAD

**Discussion by bunchofcellulose** [emilk/egui#953] [Q&A]
I am making the wikipedia game using egui. I am displaying the article, but I want the links to be displayed a buttons, which all call a method with their text, so I can render that page. I dont want all the text to be a label. How can I do that?

**Comment by parasyte** [comment]
How are you displaying the article? Is it rendering the HTML to a texture, or are you parsing the HTML into individual `egui` widgets? If it's the former, I would probably recommend switching to the latter so that you have control over how things like links are displayed.

I have a project using [`kuchiki`](https://docs.rs/kuchiki) to parse HTML, and it suits my needs well enough. I think the API has everything you need to walk the entire DOM tree, but you may not even need to touch every element for your use case.

**Comment by bunchofcellulose** [comment]
Well I am parsing the html to text and links and then converting them into egui objects and then displaying

**Reply by parasyte** [reply]
Ok, I guess I need a little more context on what the issue is and why you want a button inside a label. To me, it sounds like you are fully able to create egui buttons (or links) for each of the HTML links in the article. But I don't see where the labels come into the picture.

**Comment by emilk** [comment]
``` rust
ui.horizontal_wrapped(|ui| {
    ui.spacing_mut().item_spacing.x = 0.0;
    ui.label("Mix text and ");
    ui.hyperlink_to("hyperlinks", "https://www.github.com/emilk/egui");
    ui.label(" or ");
    if ui.button("buttons").clicked() {
        …
    }
});
```
